frontend/src/features/admin-form/create/logic/components/LogicContent/EditLogicBlock/EditCondition/ThenShowBlock.tsx

```diff
@@ -70,13 +74,69 @@ export const ThenShowBlock = ({
       : 'show'
   }, [logicTypeValue])
 
+  const [deletedFieldsCount, setDeletedFieldsCount] = useState(0)
+
+  /**
+   * Compute whether any/all fields in the show fields are deleted, then run
+   * effect to delete fields on render and show appropriate error/infobox.
+   * useWatch here to avoid infinite re-render (since if there are deleted
+   * fields, we always reset the value of the show).
+   */
+  const showValueWatch = useWatchDependency(watch, 'show')
+
+  useEffect(() => {
+    if (
+      logicTypeValue !== LogicType.ShowFields ||
+      !showValueWatch.value ||
```

Comment:
/nit Since `showValueWatch.value` is an array, we could test the for its length here too and save the processing below:
```suggestion
      !showValueWatch.value || !showValueWatch.value.length ||
```

It's inconsequential btw, since an empty array will give the correct behaviour with `0-0=0` later on in the function.

Reply:
Actually I found a bug related to this, i didn't realize `[]` is truthy (come on js, really??) so actually the behavior was not what I intended. Changing to a length check fixed this!

Reply:
Of course it's truthy! it's an object, all objects are truthy! :D